src/metabase/notification/send.clj

```diff
@@ -188,10 +181,47 @@
       (prometheus/observe! :metabase-notification/total-duration-ms {:payload-type payload_type} total-time))
     nil))
 
+(defn create-notification-dispatcher
+  "Create a thread pool for sending notifications.
+  There can only be one notification with the same id in the queue.
+  - if a notification of the same id is already in the queue, then replace it
+  - if a notification doesn't have id, put it into queue regardless (used to send unsaved notifications)"
+  [pool-size]
+  (let [ids-queue        (java.util.concurrent.LinkedBlockingQueue.)
```

Comment:
Curious around this choice - Array queue is typically better in every situation on modern hardware. Lower latency, less garbage produced. Not a big deal, just wondering if I'm missing something.

Reply:
Oh, you probably don't want it to be bounded?

I guess with our toaster model, it's hard to guess the right capacity for all users. 

My experience has revolved around situations where you want predictable memory usage and performance, but in this case I guess we can sacrifice a lot of that here since the volume won't be crazy and saving a "knob" is big!

We definitely don't want to do linear probing `(.contains ...)` with a linked list though, the pointer chasing makes that especially expensive on a large queue.

Reply:
Thanks for asking, one reason I chose this is because after this we'll want to implement priority queue, which I plan to replace this with [PriorityBlockingQueue](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/PriorityBlockingQueue.html)